Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -94,4 +141,11 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
             checkoutAttemptId = nil
         }
     }
+    
+    private func startNextTimer() {
+        batchTimer?.invalidate()
```

Comment:
Idea: maybe move the `invalidate()` to the `willSet` of the `batchTimer` this way we can make sure that it's always invalidated if a new timer is started :)